### PR TITLE
add link to list of possible dialects in the 'Using Hibernate orm and JPA' guide

### LINKED ISSUE #1564: Add link to list of possible dialects in the "USING HIBERNATE ORM AND JPA" guide
When using the MariaDB driver to connect to a MySQL 5 DB, you must explicitly specify the dialect using `quarkus.hibernate-orm.dialect=org.hibernate.dialect.MySQL5Dialect` in your application.properties, otherwise you get errors as: 

```
2019-03-18 17:50:08,702 WARN  [org.hib.too.sch.int.ExceptionHandlerLoggedImpl] (main) GenerationTarget encountered exception accepting command : Error executing DDL "create sequence known_fruits_id_seq start with 4 increment by 1" via JDBC Statement: org.hibernate.tool.schema.spi.CommandAcceptanceException: Error executing DDL "create sequence known_fruits_id_seq start with 4 increment by 1" via JDBC Statement
    at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.accept(GenerationTargetToDatabase.java:67)
    at org.hibernate.tool.schema.internal.SchemaCreatorImpl.applySqlString(SchemaCreatorImpl.java:440)
    at org.hibernate.tool.schema.internal.SchemaCreatorImpl.applySqlStrings(SchemaCreatorImpl.java:424)
    at org.hibernate.tool.schema.internal.SchemaCreatorImpl.createFromMetadata(SchemaCreatorImpl.java:290)
    at org.hibernate.tool.schema.internal.SchemaCreatorImpl.performCreation(SchemaCreatorImpl.java:166)
    at org.hibernate.tool.schema.internal.SchemaCreatorImpl.doCreation(SchemaCreatorImpl.java:135)
    at org.hibernate.tool.schema.internal.SchemaCreatorImpl.doCreation(SchemaCreatorImpl.java:121)
    at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:155)
    at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:72)
    at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:309)
    at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
    at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:80)
    at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:68)
    at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:79)
    at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:54)
    at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:114)
    at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:71)
    at io.quarkus.hibernate.orm.runtime.HibernateOrmTemplate.startAllPersistenceUnits(HibernateOrmTemplate.java:88)
    at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits12.deploy(Unknown Source)
    at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
    at io.quarkus.runtime.Application.start(Application.java:93)
    at io.quarkus.runtime.Application.run(Application.java:198)
    at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.sql.SQLSyntaxErrorException: (conn=95) You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'sequence known_fruits_id_seq start with 4 increment by 1' at line 1
    at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:236)
    at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.getException(ExceptionMapper.java:165)
    at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:238)
    at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:356)
    at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:407)
    at io.agroal.pool.wrapper.StatementWrapper.execute(StatementWrapper.java:221)
    at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.accept(GenerationTargetToDatabase.java:54)
    ... 22 more
```

In https://quarkus.io/guides/hibernate-orm-guide it would be nice to have a link to the possible values for the `quarkus.hibernate-orm.dialect` property (I found those in https://docs.jboss.org/hibernate/core/4.3/javadocs/org/hibernate/dialect/package-summary.html)

### PR DESCRIPTION
**Pull request by machi1990**
Fixes #1564 

@gastaldi @gsmet

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM. Can you confirm that the docs link matches the Hibernate version in Quarkus?

**Review by gastaldi — COMMENTED** [review]
It would be nice if the `${hibernate-orm.version}` property could be used in the link somehow

**Review by gastaldi — CHANGES_REQUESTED** [review]
https://docs.jboss.org/hibernate/orm/6.0/javadocs/org/hibernate/dialect/package-summary.html seems to match the version used

**Comment by machi1990** [comment]
> https://docs.jboss.org/hibernate/orm/6.0/javadocs/org/hibernate/dialect/package-summary.html seems to match the version used

I searched under `https://docs.jboss.org/hibernate/core/` and I could not see the current one being listed. Thanks for the link.  I have updated the PR.

**Comment by gastaldi** [comment]
@machi1990 great, but it seems the PR is still using the old link. Can you double check?

**Comment by machi1990** [comment]
> @machi1990 great, but it seems the PR is still using the old link. Can you double check?

👍 Should be good now.

**Review by gastaldi — APPROVED** [review]

**Comment by machi1990** [comment]
> It would be nice if the `${hibernate-orm.version}` property could be used in the link somehow

 With the docs only listing `major.minor` versions only e.g `5.4` or `6.0`, I am not sure how to do it since `${hibernate-orm.version}` is `5.4.2.Final`.

**Review by gsmet — CHANGES_REQUESTED** [review]
Please do not merge yet, I need to think a bit about this one.

**Comment by Sanne** [comment]
Whe're not using Hibernate ORM v.6 yet in Quarkus. Could you amend the link to point to v `5.4`?

@gsmet could you express your concerns?

File: docs/src/main/asciidoc/hibernate-orm-guide.adoc [hunk 1]
```diff
@@ -120,7 +120,7 @@ There are optional properties useful to refine your `EntityManagerFactory` or gu
 ==== Dialect
 
 `quarkus.hibernate-orm.dialect`:: (e.g. `org.hibernate.dialect.PostgreSQL95Dialect`).
-Class name of the Hibernate ORM dialect.
+Class name of the Hibernate ORM dialect. Complete list of available dialects values can be seen https://docs.jboss.org/hibernate/orm/5.4/javadocs/org/hibernate/dialect/package-summary.html[here] in the `Class Summary` section. A concrete `dialect` value may then be specified by prefixing a Class name with `"org.hibernate.dialect."` .
```
Comment (gsmet) [inline docs/src/main/asciidoc/hibernate-orm-guide.adoc]: I would go with this:
`The complete list of bundled dialects is available in the https://docs.jboss.org/hibernate/stable/orm/javadocs/org/hibernate/dialect/package-summary.html[Hibernate ORM JavaDoc].`

I would also add a note just at the end of this Dialect paragraph saying:
```
[NOTE]
====
Not all the dialects are supported in GraalVM native executables: we currently provide driver extensions for PostgreSQL, MariaDB and Microsoft SQL Server and H2.
====
```

@machi1990 can you take care of this? Thanks.

Comment (machi1990) [reply docs/src/main/asciidoc/hibernate-orm-guide.adoc]: Done. 

I dropped the extra `"and"` from the note phrase`..PostgreSQL, MariaDB and Microsoft SQL Server and H2.`

Comment (gsmet) [reply docs/src/main/asciidoc/hibernate-orm-guide.adoc]: Good call, I put H2 at the beginning then at the end and forgot to remove the `and`.

Thanks for taking care of it, merging.

**Review by gsmet — APPROVED** [review]

**Pull request merged**